lib/active_record/virtual_attributes/virtual_fields.rb

```diff
@@ -221,6 +221,12 @@ def select(*fields)
         super
       end
 
+      # From ActiveRecord::QueryMethods
+      def build_left_outer_joins(manager, outer_joins, *rest)
+        outer_joins = klass.replace_virtual_fields(outer_joins)
+        super if outer_joins.present?
```

Comment:
do we need to call `super` regardless?

Reply:
we are modifying `outer_joins`. We could be adding, keeping the same, or removing it.

We only want to call `build_left_outer_joins` if we actually have an `outer_joins` present.

[calling code](https://github.com/rails/rails/blob/58256db50750781016f5ea6d2dd6cbc72096062b/activerecord/lib/active_record/relation/query_methods.rb#L1034) only calls unless `outer_joins.empty?`